datumaro/components/launcher.py

```diff
@@ -36,12 +38,20 @@ def __init__(self, extractor, launcher, batch_size=1):
 
     def __iter__(self):
         for batch in take_by(self._extractor, self._batch_size):
-            inputs = np.array([np.atleast_3d(item.media.data) for item in batch])
+            inputs = []
+            for item in batch:
+                if not isinstance(item.media, Image):
+                    raise MediaTypeError(
+                        f"Media type should be Image, Current type={type(item.media)}"
+                    )
+                inputs.append(np.atleast_3d(item.media.data))
```

Comment:
Is there any reason to restrict `item.media` type to `Image`? This change may affect other parts.

Reply:
Fix. I made `type_check` function for searcher launcher as your suggestion.